PowerEditor/src/NppCommands.cpp

```diff
@@ -1546,6 +1546,12 @@ void Notepad_plus::command(int id)
 			_pEditView->execute(SCI_ENDUNDOACTION);
 			break;
 
+		case IDM_EDIT_REMOVE_ANY_DUP_LINES:
+			_pEditView->execute(SCI_BEGINUNDOACTION);

```

Comment:
@donho or anyone:
Is this needed to be added between L 1549 and 1550?:
```
std::lock_guard<std::mutex> lock(command_mutex);
```

Reply:
@sasumner 
> Is this needed to be added between L 1549 and 1550?:
std::lock_guard<std::mutex> lock(command_mutex);

Why do you think we need this line?

Reply:
@donho 

I was hoping you might tell me the answer to that. :-)

I noticed that the same code is used before sorting, which is similar to this new operation, see https://github.com/notepad-plus-plus/notepad-plus-plus/blob/0a209393347a362ada92678c2316721c3a411986/PowerEditor/src/NppCommands.cpp#L578

This new one is more similar to the process for sorting and less similar to the "remove consecutive duplicate lines" feature.

Reply:
Scintilla is not thread safe, so for some long operation, `std::lock_guard<std::mutex> lock` is used with `{` and `}` for guarantee no race condition. In this case, I think we can do without it.